src/cysignals/tests_helper.c

```diff
@@ -112,3 +116,22 @@ void signal_pid_after_delay(int signum, pid_t killpid, long ms, long interval, i
 
 /* The same as above, but sending ``n`` signals */
 #define signals_after_delay(signum, ms, interval, n) signal_pid_after_delay(signum, getpid(), ms, interval, n)
+
+#else
+void ms_sleep(long ms)
+{
+    Sleep(ms);
+}
+
+void signal_after_delay(int signum, long ms, long interval)
```

Comment:
Can you check what `raise()` does in a multi-threaded application on Windows? If it sends the signal to the whole process (all threads), then you could use threading instead of `fork()`.

Note that this won't work on Linux since `raise()` signals only the calling thread.

Reply:
It's supposed to kill the program. So i don't think multithreading with `raise` will be the solution.